integration-libs/cds/src/components/recent-searches/recent-searches.service.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { interval, Observable, of, ReplaySubject } from 'rxjs';
+import { concatMap, endWith, take, takeWhile } from 'rxjs/operators';
+
+@Injectable()
+export class RecentSearchesService {
+  private recentSearchesSource = new ReplaySubject<string[]>();
+  private apiAvailability = false;
+  public get recentSearches$(): Observable<string[]> {
+    this.addRecentSearchesListener();
+    return this.recentSearchesSource.asObservable();
+  }
+  checkAvailability() {
+    return interval(150).pipe(
+      concatMap((_) => of((<any>window).Y_TRACKING)),
```

Comment:
@Platonn do You have any idea what we can use in that case? If we know the Y_TRACKING (recentSeraches) is on `window`. Maybe some proposition?

Reply:
I changed to `windowRef from @spartacus/core` instead of `window` -> What do you think?

Reply:
WindowRef.nativeWindow has a type of `Window | undefined`. On the server it will return `undefined`.  So you need to add `?` before `.Y_TRACKING)`

Reply:
I'm not sure i also understand correctly the propose for` endWith(true)`. If after 5th attempt Y_TRACKING will still be `undefined`, it will return `true`. Thats mean that we will try to execute this code 
```
    const recentPhrases = (<any>(
            this.winRef.nativeWindow
          )).Y_TRACKING?.recentSearches?.getPhrases();
```

but we don't have to, if we already know that Y_TRACKING is `undefined`.